packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForFailed().should.equal(50);
+        });
+
+        it('should be possible update with an array of transactions', () => {
+
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus0 = new TxStatus();
+            clock.tick(25);
+            const txStatus1 = new TxStatus();
+            txStatus0.SetStatusSuccess();
+            clock.tick(100);
+            txStatus1.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished([txStatus0, txStatus1]);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(2);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(75);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(75);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(175);
+            transactionStatisticsCollector.getMinLatencyForSuccessful().should.equal(25);
+            transactionStatisticsCollector.getMaxLatencyForSuccessful().should.equal(100);
+            transactionStatisticsCollector.getTotalLatencyForSuccessful().should.equal(125);
+        });
+    });
+
+    describe('Aggregating multiple TransactionStatisticsCollector results', () => {
+
+        it('should be possible to merge TransactionStatisticsCollector results into a single aggregation', () => {
+
+            const transactionStatisticsCollector0 = new TransactionStatisticsCollector();
+            transactionStatisticsCollector0.activate();
+
+            const transactionStatisticsCollector1 = new TransactionStatisticsCollector();
+            transactionStatisticsCollector1.activate();
+
+            clock.tick(50);
+            const txStatus0 = new TxStatus();
+            clock.tick(25);
+            const txStatus1 = new TxStatus();
+            txStatus0.SetStatusSuccess();
+            clock.tick(100);
+            txStatus1.SetStatusSuccess();
+
+            transactionStatisticsCollector0.txFinished(txStatus0);
+            transactionStatisticsCollector1.txFinished(txStatus1);
+
+            const mergedCollector = TransactionStatisticsCollector.mergeCollectorResults([transactionStatisticsCollector0, transactionStatisticsCollector1]);
+
+            mergedCollector.getTotalSuccessfulTx().should.equal(2);
+            mergedCollector.getFirstCreateTime().should.equal(50);
```

Comment:
It might be better to use the `txStatus.GetXY()` return values for the expected values, rather than literal numbers. That will clearly indicate the source of the expected value. (Only applies if a metric is derived from txStat metrics.)